salt/state.py

```diff
@@ -883,19 +883,31 @@ def _run_check_onlyif(self, low_data, cmd_opts):
         else:
             low_data_onlyif = low_data['onlyif']
         for entry in low_data_onlyif:
-            if not isinstance(entry, six.string_types):
+            if isinstance(entry, six.string_types):
+                cmd = self.functions['cmd.retcode'](
+                    entry, ignore_retcode=True, python_shell=True, **cmd_opts)
+                log.debug('Last command return code: %s', cmd)
+                if cmd != 0 and ret['result'] is False:
+                    ret.update({'comment': 'onlyif condition is false',
+                                'skip_watch': True,
+                                'result': True})
+                elif cmd == 0:
+                    ret.update({'comment': 'onlyif condition is true', 'result': False})
+            elif isinstance(entry, dict):
+                if 'fun' not in entry:
+                    ret['comment'] = 'no `fun` argument in onlyif: {0}'.format(entry)
+                    log.warning(ret['comment'])
+                    return ret
+                result = self.functions[entry.pop('fun')](**entry)
```

Comment:
nice 👍 i'd suggest adding a check for __context__['retcode'] and taking that over the return result if set

Reply:
ahh good point.

Reply:
going to make it only care about if retcode is anything but 0.

Reply:
since otherwise, we would always be 0, because the salt minion sets it to 0 if nothing is set.

Reply:
Done!